### PR TITLE
Conveyor mapper updates

### PR DESCRIPTION
**Pull request by rbogren-brock**
Updates to several fields for the conveyor-HDF mapper. Changes made to meet the SwAP use case.

### CODE REVIEW

File: libs/hdf-converters/package.json [hunk 1]
```diff
@@ -63,7 +63,7 @@
     "quicktype": "^15.0.260",
     "ts-jest": "^29.1.0",
     "ts-node": "^10.0.0",
-    "typedoc": "^0.25.0"
+    "typedoc": "^0.24.1"
```
Comment (Amndeep7) [inline libs/hdf-converters/package.json:66]: why the regression in version number?

File: libs/hdf-converters/src/conveyor-mapper.ts [hunk 2]
```diff
@@ -10,16 +10,23 @@ const CONVEYOR_MAX_SCORE = 1000;
 enum scannerType {
   Moldy = 'Moldy',
   Stigma = 'Stigma',
-  CodeQuality = 'CodeQuality'
+  CodeQuality = 'CodeQuality',
+  Clamav = 'Clamav'
```
Comment (Amndeep7) [inline libs/hdf-converters/src/conveyor-mapper.ts:14]: would prefer to keep the capitalization of the original application: ClamAV

File: libs/hdf-converters/src/conveyor-mapper.ts [hunk 3]
```diff
@@ -10,16 +10,23 @@ const CONVEYOR_MAX_SCORE = 1000;
 enum scannerType {
```
Comment (Amndeep7) [inline libs/hdf-converters/src/conveyor-mapper.ts:10]: might be worthwhile to order these alphabetically at this point

File: libs/hdf-converters/src/conveyor-mapper.ts [hunk 4]
```diff
@@ -67,7 +77,9 @@ function mapSha2Filename(
   }
   return shaMappings;
 }
-
+/*
+Builds the code_desc field using values from data. Varies based on service
```
Comment (Amndeep7) [inline libs/hdf-converters/src/conveyor-mapper.ts:81]: i appreciate the comments

File: libs/hdf-converters/src/conveyor-mapper.ts [hunk 5]
```diff
@@ -142,18 +155,21 @@ function preprocessObject(
...
+*/ 
 function controlMappingConveyor(): MappedTransform<
   ExecJSON.Control & ILookupPath,
   ILookupPath
 > {
   return {
     id: {path: 'sha256'},
-    title: {path: 'filename'},
-    desc: '',
+    title: {path: 'filename'},  // Should be heur_id from conveyor
```
Comment (Amndeep7) [inline libs/hdf-converters/src/conveyor-mapper.ts:167]: there seems to be a discrepancy between the newly added comment and what attribute we've been using here.  which is correct?

File: libs/hdf-converters/src/conveyor-mapper.ts [hunk 6]
```diff
@@ -245,10 +264,12 @@ export class ConveyorResults {
       Object.entries(
         _.get(this.data, 'api_response.results') as Record<string, unknown>
       ) as [string, Record<string, unknown>][]
-    ).map(([scannerName, scannerData]) => [
-      scannerName,
-      new ConveyorMapper(scannerData, this.data, scannerName).toHdf()
-    ]);
+    ).map(([scannerName, scannerData]) => {
```
Comment (Amndeep7) [inline libs/hdf-converters/src/conveyor-mapper.ts:267]: why use a function body when we immediately return a single expression?

**Review by Amndeep7 — CHANGES_REQUESTED** [review]

**Review by Amndeep7 — CHANGES_REQUESTED** [review]
also now that I've enabled the workflows, the linter is bad at you.  you should be able to just run `yarn run lint` and have it autofixed.

**Comment by rbogren-brock** [comment]
Ran linter and addressed comments. No disagreements with any of the suggestions.

File: libs/inspecjs/package.json [hunk 7]
```diff
@@ -15,7 +15,7 @@
     "jest": "^27.0.6",
     "quicktype": "^15.0.260",
     "ts-jest": "^29.1.0",
-    "typedoc": "^0.25.0"
+    "typedoc": "^0.24.1"
```
Comment (Amndeep7) [inline libs/inspecjs/package.json:18]: made the change in the package.json for hdf-converters but not here

File: yarn.lock
Comment (Amndeep7) [inline yarn.lock:1]: if we're intentionally not making any changes to the package.json, (i.e. leaving typedoc at 0.25.x) then let's not make any changes to the yarn lock file either.

File: libs/hdf-converters/src/conveyor-mapper.ts
Comment (Amndeep7) [inline libs/hdf-converters/src/conveyor-mapper.ts:1]: would be great if you could add another test or modify the samples to also include clamav results

**Review by Amndeep7 — CHANGES_REQUESTED** [review]
would appreciate it if you could give this pr a once-over as well so we can merge it in too: https://github.com/mitre/saf/pull/1488/files

**Review by Amndeep7 — APPROVED** [review]

**Pull request merged**